canonicalwebteam/blog/wordpress_api.py

```diff
@@ -19,71 +21,78 @@ def process_response(response):
     return response.json()
 
 
-def build_get_articles_url(
-    tags=[],
-    per_page=12,
-    page=1,
-    tags_exclude=[],
-    exclude=[],
-    categories=[],
-    sticky="",
-    groups=[],
-    after="",
-    before="",
-    author="",
-):
+def build_url(endpoint, params={}):
     """
     Build url to fetch articles from Wordpress api
-    :param tags: Array of tag ids to fetch articles for
-    :param per_page: Articles to get per page
-    :param page: Page number to get
-    :param tags_exclude: Array of IDs of tags that will be excluded
-    :param exclude: Array of article IDs to be excluded
-    :param category: Array of categories, which articles
-        should be fetched for
-    :param sticky: string 'true' or 'false' to only get featured articles
-    :param before: ISO8601 compliant date string to limit by date
-    :param after: ISO8601 compliant date string to limit by date
+    :param endpoint: The REST endpoint to fetch data from
+    :param params: Dictionary of parameter keys and their values
 
     :returns: URL to Wordpress api
     """
-    url = (
-        f"{API_URL}/posts?per_page={per_page}"
-        f"&tags={','.join(str(id) for id in tags)}"
-        f"&page={page}"
-        f"&group={','.join(str(id) for id in groups)}"
-        f"&tags_exclude={','.join(str(id) for id in tags_exclude)}"
-        f"&categories={','.join(str(id) for id in categories)}"
-        f"&exclude={','.join(str(id) for id in exclude)}"
-        f"&author={author}&_embed"
+
+    def list_to_comma_separated_string(key, value):
+        if type(value) is list:
+            return (key, (",".join(str(item) for item in value)))
+        else:
+            return (key, value)
+
+    params = {key: value for key, value in params.items() if value}
+    params = dict(
+        map(lambda kv: list_to_comma_separated_string(*kv), params.items())
     )
-    if sticky != "":
-        url = url + f"&sticky={sticky}"
-    if before != "":
-        url = url + f"&before={before}"
-    if after != "":
-        url = url + f"&after={after}"
 
-    return url
+    params = urlencode(params)
+
+    return (
+        f"{API_URL}/{endpoint}?{params}&_embed"
+        if params
+        else f"{API_URL}/{endpoint}&_embed"
+    )
 
 
-def get_articles_with_metadata(**kwargs):
+def get_articles(
+    tags=[],
+    tags_exclude=[],
+    exclude=[],
+    categories=[],
+    sticky="",
```

Comment:
It's common practice to set optional arguments to `None`, unless you actually intend to use a different value. So unless you intend to have `sticky=""` in your resulting API query, I think `None` would be a lot more intuitive.

Reply:
Aight. Changing this makes the `for loop` approach easy.